vars/piperStageWrapper.groovy

```diff
@@ -76,20 +78,19 @@ private void executeStage(script, originalStage, stageName, config, utils) {
 
         // First, check if a global extension exists via a dedicated repository
         if (globalExtensions) {
-            Script globalInterceptorScript = load(globalInterceptorFile)
             echo "[${STEP_NAME}] Found global interceptor '${globalInterceptorFile}' for ${stageName}."
             // If we call the global interceptor, we will pass on originalStage as parameter
             body = {
-                globalInterceptorScript(script: script, originalStage: body, stageName: stageName, config: config)
+                callInterceptor(script, globalInterceptorFile, originalStage, stageName, config)
```

Comment:
Just for consistency I would use

```suggestion
                callInterceptor(script, globalInterceptorFile, body, stageName, config)
```

Reply:
I remember wondering about that. I wasn't sure which contents of "body" is visible in the closure, when it is invoked later. I didn't want to produce an endless loop when the closure invokes itself. The version in runAsStage() had it like in my PR. So I take it "body" (inside the closure) resolves to whatever state it has at the time of constructing the closure?

Reply:
To anyone reading later: Any changes to the local variable "body" will become visible in the closure as well. So this would indeed introduce an infinite loop. "body" will *not* be captured in the closure in its state when the closure is constructed. This explains also a TODO farther below in the code.